sjsonnet/src/sjsonnet/BaseCharRenderer.scala

```diff
@@ -136,20 +145,20 @@ class BaseCharRenderer[T <: upickle.core.CharOps.Output]
   }
 
 
-  def visitString(s: CharSequence, index: Int) = {
+  def visitString(s: CharSequence, index: Int): T = {
 
     if (s eq null) visitNull(index)
     else visitNonNullString(s, index)
   }
 
-  def visitNonNullString(s: CharSequence, index: Int) = {
+  private def visitNonNullString(s: CharSequence, index: Int) = {
     flushBuffer()
-    upickle.core.RenderUtils.escapeChar(unicodeCharBuilder, elemBuilder, s, escapeUnicode)
+    upickle.core.RenderUtils.escapeChar(null, elemBuilder, s, escapeUnicode)
```

Comment:
It looks like this `unicodeCharBuilder` parameter is accepted by `escapeChar` but isn't actually used by that code. Checking the history of that class in upickle, it looks like it's _never_ been used by that method.

I think it's probably fine to pass `null` here.

Is the intent here just to save a small amount of unnecessary allocation per `BaseCharRenderer` construction?

Reply:
I think? It sometimes feel like a lot of microoptimizations are used and I'm always a bit wary of changing things